src/vos/vos_internal.h

```diff
@@ -427,6 +427,27 @@ int
 vos_dtx_check_availability(daos_handle_t coh, uint32_t entry,
 			   daos_epoch_t epoch, uint32_t intent, uint32_t type);
 
+/**
+ * Get entry DTX state.
+ *
+ * \param entry		[IN]	DTX local id
+ *
+ * \return		DTX_ST_COMMITTED, DTX_ST_PREPARED or
+ *			DTX_ST_ABORTED.
+ */
+static inline unsigned int
+vos_dtx_ent_state(uint32_t entry)
+{
+	switch (entry) {
+	case DTX_LID_COMMITTED:
+		return DTX_ST_COMMITTED;
+	case DTX_LID_ABORTED:
+		return DTX_ST_ABORTED;
+	default:
+		return DTX_ST_PREPARED;
```

Comment:
If it is only for aggregation case, it may work; but be as general function, it is not correct. If the "entry" is neither "DTX_LID_COMMITTED" nor "DTX_LID_ABORTED", then it can be 'prepared', but also may be 'committable'. If we regard it as 'prepared' always, it may misguide the non-aggregation caller.

Reply:
As I suggested, could you implement a real 'get entry dtx state' function to replace this one later?

Reply:
Honestly, it is not easy to implement that unless:
1) DTX logic is totally removed from VOS, or
2) Introduce some kinds of upcall to trigger DTX refresh inside VOS.

Both of them may not happen in a short time. Currently, I would suggest to add some comment in the function to indicate that it is only can be used for VOS aggregation temporarily.